systest/plugin/integration_test.go

```diff
@@ -0,0 +1,62 @@
+//go:build integration
+
+/*
+ * Copyright 2023 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/dgraph-io/dgraph/dgraphtest"
+	"github.com/dgraph-io/dgraph/x"
+)
+
+type PluginTestSuite struct {
+	suite.Suite
+	dc dgraphtest.Cluster
+	pfiEntry PluginFuncInp
+}
+
+func (psuite *PluginTestSuite) SetupTest() {
+	psuite.dc = dgraphtest.NewComposeCluster()
+}
+
+func (psuite *PluginTestSuite) TearDownTest() {
+	t := psuite.T()
+	gcli, cleanup, err := psuite.dc.Client()
+	require.NoError(t, err)
+	defer cleanup()
+	require.NoError(t, gcli.LoginIntoNamespace(context.Background(),
+		dgraphtest.DefaultUser, dgraphtest.DefaultPassword, x.GalaxyNamespace))
+	require.NoError(t, gcli.DropAll())
+}
+
+func (psuite *PluginTestSuite) Upgrade() {
+	// Not implemented for integration tests
+}
+
+func TestPluginTestSuite(t *testing.T) {
+	var psuite PluginTestSuite
+	for _, e := range pfiArray {
+		psuite.pfiEntry = e
+		suite.Run(t, &psuite)
```

Comment:
we should check for the exit code here. So that if the tests fail, we do not execute the next iteration. For example, https://github.com/dgraph-io/dgraph/pull/8792/files#diff-ec415551802feb32bd210f935f5bec664608e6c4e6613b56d08545368619d30fR61